src/firebase-config-common.ts

```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type CommonDocumentData = { [field: string]: any };
+/** An interface for `FirestoreDataConverter` common to both frontend firebase and admin firebase. */
+export interface CommonFirestoreDataConverter<T> {
+  toFirestore(modelObject: T): CommonDocumentData;
+  fromFirestore(snapshot: { data(): CommonDocumentData }): T;
+}
```

Comment:
Firestore has almost the exact same TS typedef on frontend and in admin, but they are slightly different so it's not structurally the same. Therefore, I have to exact this minimal interface that satisfy both.